### PR TITLE
Store FeeCalculator with blockhash in nonce accounts

### LINKED ISSUE #7967: Durable nonce transactions have an effectively unpredictable fee schedule
#### Problem

A typical transaction is charged fees based on the value of its `recent_blockhash` field.  The `FeeCalculator` and corresponding blockhash are stored in the working bank's `RecentBlockhashQueue`.  Since a durable nonce transaction's `recent_blockhash` is necessarily no longer in the bank's `RecentBlockhashQueue` the initial implementation simply takes the `FeeCalculator` of the working bank's `last_blockhash`.  This introduces unpredictablity should the fee schedule move significantly between the transaction being created and submitted to the network, which is not ideal.

#### Proposed Solution

1. Store the fee calculator with the account's metadata.  This may incentivize stocking up on low-fee nonce accounts for use in the event of high fees.
1. Leave it as is and accept this inconsistency as a cost of using durable nonces

tag: @garious @rob-solana @mvines

### PR DESCRIPTION
**Pull request by t-nelson**
#### Problem

Nonce accounts don't store the associated `FeeCalculator` alongside the blockhash making fees for transactions that use them unpredictable.

#### Summary of Changes

Store `FeeCalculator`s with the blockhash in nonce accounts
Query the stored `FeeCaculator` rather than the most recent one when charging fees
Fix up tests

Fixes #7967

### CODE REVIEW

**Comment by t-nelson** [comment]
@mvines I'm still fighting one test here, but getting a first pass at review would be great!

Offending test is `bank::tests::test_durable_nonce_transaction` if there's an extra set of :eyes: free

File: cli/src/nonce.rs [hunk 1]
```diff
@@ -569,10 +569,13 @@ pub fn process_show_nonce_account(
         match data {
             Some(ref data) => {
                 println!("Nonce: {}", data.blockhash);
+                println!("Fees*: {}", data.fee_calculator.lamports_per_signature);
                 println!("Authority: {}", data.authority);
+                println!(" * Lamports per signature");
             }
```
Comment (mvines) [inline cli/src/nonce.rs]:
```suggestion
                println!("Fee: {} lamports per signature", data.fee_calculator.lamports_per_signature);
                println!("Authority: {}", data.authority);
            }
```

**Comment by mvines** [comment]
The changes make sense to me, mostly just shuffling code around to adapt to the new nonce account fee_calculator field.  Not sure about that test, I'd need to actually play with it I think to spot what's going wrong

**Comment by t-nelson** [comment]
:pray: 

RE Test: Looks like genesis and bank[1] have fees = 0, is the problem.  Not sure when that was introduced... still digging.

**Comment by t-nelson** [comment]
Actually, this makes sense. Since the nonce accounts were previously using fees from the most recent block, they were already primed by the time transactions start executing since this test waits until the `BlockhashQueue` is full beforehand. Now the nonce account is dragging Genesis fee rate ahead with it and exposing the "issue".  As for the 0 fees in banks 0 and 1, this makes sense as well, because we don't actually calculate fees until `FeeRateGovernor::new_derived()` is called, which happens in `Bank::new_from_parent()`

tldr;  Assuming we're cool with TXs citing the blockhash of banks 0 and 1 being free, which I believe is legacy behavior, I can just idle the test for two blocks as a fix.

**Comment by mvines** [comment]
> tldr; Assuming we're cool with TXs citing the blockhash of banks 0 and 1 being free, which I believe is legacy behavior, I can just idle the test for two blocks as a fix.

Fun!  Yeah that's fine.  You cannot get an external transaction into bank 0, and best of luck getting into bank 1.  Bragging rights forever if you do it.

**Comment by t-nelson** [comment]
> Bragging rights forever if you do it.

And as many initialized nonce accounts as I can manage!

**Comment by mvines** [comment]
@t-nelson - heads up that this'll need a manual rebase for v1.0
